src/examples/client/simpleStreamableHttp.ts

```diff
@@ -24,50 +25,79 @@ async function main(): Promise<void> {
   const transport = new StreamableHTTPClientTransport(
     new URL('http://localhost:3000/mcp')
   );
+  let supportsStandaloneSse = false;
 
   // Connect the client using the transport and initialize the server
   await client.connect(transport);
+  console.log('Connected to MCP server');
+  // Open a standalone SSE stream to receive server-initiated messages
+  console.log('Opening SSE stream to receive server notifications...');
+  try {
+    await transport.openSseStream();
+    supportsStandaloneSse = true;
+    console.log('SSE stream established successfully. Waiting for notifications...');
+  }
+  catch (error) {
+    console.error('Failed to open SSE stream:', error);
+  }
+
+  // Set up notification handlers for server-initiated messages
   client.setNotificationHandler(LoggingMessageNotificationSchema, (notification) => {
     console.log(`Notification received: ${notification.params.level} - ${notification.params.data}`);
   });
+  client.setNotificationHandler(ResourceListChangedNotificationSchema, async (_) => {
+    console.log(`Resource list changed notification received!`);
+    const resourcesRequest: ListResourcesRequest = {
+      method: 'resources/list',
+      params: {}
+    };
+    const resourcesResult = await client.request(resourcesRequest, ListResourcesResultSchema);
+    console.log('Available resources count:', resourcesResult.resources.length);
+  });
 
-
-  console.log('Connected to MCP server');
   // List available tools
-  const toolsRequest: ListToolsRequest = {
-    method: 'tools/list',
-    params: {}
-  };
-  const toolsResult = await client.request(toolsRequest, ListToolsResultSchema);
-  console.log('Available tools:', toolsResult.tools);
+  try {
+    const toolsRequest: ListToolsRequest = {
+      method: 'tools/list',
+      params: {}
+    };
+    const toolsResult = await client.request(toolsRequest, ListToolsResultSchema);
+    console.log('Available tools:', toolsResult.tools);
 
-  // Call the 'greet' tool
-  const greetRequest: CallToolRequest = {
-    method: 'tools/call',
-    params: {
-      name: 'greet',
-      arguments: { name: 'MCP User' }
-    }
-  };
-  const greetResult = await client.request(greetRequest, CallToolResultSchema);
-  console.log('Greeting result:', greetResult.content[0].text);
+    if (toolsResult.tools.length === 0) {
+      console.log('No tools available from the server');
```

Comment:
Nit: I suggest moving this tool calling code into a function or something, so this branch could be an early return (making things more readable).